inference-engine/src/inference_engine/cpp/ie_infer_request.cpp

```diff
@@ -137,10 +277,14 @@ void InferRequest::SetCompletionCallbackImpl(std::function<void()> callback) {
         CATCH_IE_EXCEPTION_RETURN(INFER_CANCELLED, InferCancelled)
 
 
-void InferRequest::SetCompletionCallbackImpl(std::function<void(InferRequest, StatusCode)> callback) {
+void InferRequest::SetCompletionCallbackImpl(std::function<void(InferRequest, StatusCode)> callbackToSet) {
+    if (actual) {
+        IE_THROW() << "Cannot set callback with InferRequest";
```

Comment:
@apankratovantonp please, correct me if old behavior was different.

Reply:
Seems you are missed specialization:
```c++
emplate <>
class CompletionCallbackWrapper<std::function<void(InferRequest, StatusCode)>> : public ICompletionCallbackWrapper {
    std::function<void(InferRequest, StatusCode)> lambda;

public:
    explicit CompletionCallbackWrapper(const std::function<void(InferRequest, InferenceEngine::StatusCode)>& lambda)
        : lambda(lambda) {}

    void call(InferenceEngine::IInferRequest::Ptr request, InferenceEngine::StatusCode code) const noexcept override {
        lambda(InferRequest(request), code);
    }
};
```

Reply:
thanks, fixed!